app/sidero-controller-manager/api/v1alpha2/server_webhook.go

```diff
@@ -0,0 +1,118 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
I feel it would be great to extract that and share the validation code between v1 and v2, but I'm not sure if it goes against the best practices :man_shrugging:

Reply:
well, validation might be different across the two I guess? as we evolve the types

Reply:
It may still have common parts I guess.